bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,237 @@
+
+	// Response struct
+	var appListResponse FetchMyAppsResponse
+	responseStatusCode := -1
+
+	stillPaging := true
+	var next string
+	for stillPaging {
+
+		headers := client.headers
+
+		request, cerr := createRequest(http.MethodGet, requestURL, headers, nil)
+		if cerr != nil {
+			err = cerr
+			return
+		}
+
+		if len(next) > 0 {
+			quearryValues := request.URL.Query()
+			quearryValues.Add("next", next)
+			request.URL.RawQuery = quearryValues.Encode()
+		}
+
+		//
+		// Perform request
+		if cerr := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+			body, statusCode, err := performRequest(request)
+			if err != nil {
+				log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+				if !strings.Contains(err.Error(), "failed to perform request") {
+					log.Warnf("Response status: %d", statusCode)
+					log.Warnf("Body: %s", string(body))
+				}
+				return err
+			}
+
+			responseStatusCode = statusCode
+
+			// Parse JSON body
+			if err := json.Unmarshal([]byte(body), &appListResponse); err != nil {
+				return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+			}
+			return nil
+
+		}); cerr != nil {
+			err = cerr
+			return
+		}
+
+		logDebugPretty(appListResponse)
+
+		apps = append(apps, appListResponse.Data...)
+
+		if len(appListResponse.Paging.Next) > 0 {
+			next = appListResponse.Paging.Next
+			appListResponse = FetchMyAppsResponse{}
+		} else {
+			stillPaging = false
+		}
+
+	}
+
+	return
+}
+
+// SetSelectedAppSlug ...
+func (client *BitriseClient) SetSelectedAppSlug(slug string) {
+	client.selectedAppSlug = slug
+}
+
+func createUploadRequest(requestMethod string, url string, headers map[string]string, filePth string) (*http.Request, error) {
+	var fContent []byte
+
+	f, err := os.Open(filePth)
+	if err != nil {
+		return nil, err
+
+	}
+
+	fContent, err = ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewReader(fContent))
+	if err != nil {
+		return nil, err
+	}
+
+	for key, value := range headers {
+		req.Header.Add(key, value)
+
+	}
+
+	return req, nil
+}
+
+func createRequest(requestMethod string, url string, headers map[string]string, fields map[string]interface{}) (*http.Request, error) {
+	b := new(bytes.Buffer)
+
+	if len(fields) > 0 {
+		err := json.NewEncoder(b).Encode(fields)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	log.Debugf("Request body: %s", string(b.Bytes()))
+
+	req, err := http.NewRequest(requestMethod, url, bytes.NewReader(b.Bytes()))
+	if err != nil {
+		return nil, err
+	}
+
+	for key, value := range headers {
+		req.Header.Add(key, value)
+
+	}
+
+	return req, nil
+}
+
+func performRequest(request *http.Request) ([]byte, int, error) {
+	client := http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		// On error, any Response can be ignored
+		return []byte{}, -1, fmt.Errorf("failed to perform request, error: %s", err)
```

Comment:
if we do not use the returned byte slice in case of error, use: return nil, ....